File: TP2/Ej4/ej4b/fragments-dinamicos/app/src/main/java/com/example/fragmentsdinamicos/MainActivity.java
package com.example.fragmentsdinamicos;

import android.os.Bundle;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    FragmentRed fragmentRed = new FragmentRed();
    FragmentGreen fragmentGreen = new FragmentGreen();
    FragmentBlue fragmentBlue = new FragmentBlue();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        getSupportFragmentManager().beginTransaction().add(R.id.fragmentContainer, fragmentRed).commit();

        Button btnRed = findViewById(R.id.btnRed);
        Button btnGreen = findViewById(R.id.btnGreen);
        Button btnBlue = findViewById(R.id.btnBlue);

        btnRed.setOnClickListener(v -> {
            getSupportFragmentManager().beginTransaction().replace(R.id.fragmentContainer, fragmentRed).commit();
        });

        btnGreen.setOnClickListener(v -> {
            getSupportFragmentManager().beginTransaction().replace(R.id.fragmentContainer, fragmentGreen).commit();
        });

        btnBlue.setOnClickListener(v -> {
            getSupportFragmentManager().beginTransaction().replace(R.id.fragmentContainer, fragmentBlue).commit();
        });
    }
}

File: tp2/ej3/ej3a-contactos/app/src/main/java/com/example/contacto/MainActivity.kt
package com.example.contacto

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.contacto.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding
    private lateinit var dbHelper: ContactDatabaseHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        enableEdgeToEdge()
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        dbHelper = ContactDatabaseHelper(this)
        val newContact1 = Contact(name = "Juan", phone = "1111")
        val newContact2 = Contact(name = "Pedro", phone = "2222")
        val newContact3 = Contact(name = "Agus", phone = "3333")
        val newContact4 = Contact(id = 1, name = "Juan2", phone = "9999")

        dbHelper.addContact(newContact1)
        dbHelper.addContact(newContact2)
        dbHelper.addContact(newContact3)

        dbHelper.updateContact(newContact4)

        dbHelper.deleteContact(2)

        val contacts = dbHelper.getAllContacts()

        val text = contacts.joinToString(separator = "\n") {
            "ID: ${it.id}, Name: ${it.name}, Phone: ${it.phone}"
        }

        binding.textView.text = text
    }
}